### ISSUE TITLE
Keras adds unconditional losses for each time a model is called on new ops.

### ISSUE THREAD

**Issue by TimZaman** [keras-team/keras#8700]
Minimal example:

```python
import tensorflow as tf
import numpy as np
import os
import keras

tf.set_random_seed(42)
np.random.seed(42)

x_np = [[0, 0], [0, 1], [1, 0], [1, 1]]
x = tf.constant(x_np, dtype=tf.float32)
y_np = [[1, 0],  [0, 1], [0, 1], [1, 0]]
yp = tf.constant(y_np, dtype=tf.float32)

x_in = keras.layers.Input(tensor=x)
y = keras.layers.Dense(4, activation='relu', kernel_regularizer=keras.regularizers.l2(0.01))(x_in)
y = keras.layers.Dense(2, activation=None)(y)

m = keras.models.Model(inputs=x_in, outputs=y)

m.compile(optimizer='adam', loss='mse', target_tensors=[yp])

for i in range(10):
  out = m(tf.zeros([4,2]))  # Stack the model on top of some input
  print(m._per_input_losses)  # You can see the input losses are increasing
```

Output:

```
{None: [<tf.Tensor 'dense_1/add:0' shape=() dtype=float32>]}
{None: [<tf.Tensor 'dense_1/add:0' shape=() dtype=float32>, <tf.Tensor 'dense_1/add:0' shape=() dtype=float32>]}
{None: [<tf.Tensor 'dense_1/add:0' shape=() dtype=float32>, <tf.Tensor 'dense_1/add:0' shape=() dtype=float32>, <tf.Tensor 'dense_1/add:0' shape=() dtype=float32>]}
{None: [<tf.Tensor 'dense_1/add:0' shape=() dtype=float32>, <tf.Tensor 'dense_1/add:0' shape=() dtype=float32>, <tf.Tensor 'dense_1/add:0' shape=() dtype=float32>, <tf.Tensor 'dense_1/add:0' shape=() dtype=float32>]}
```

Because they key of the `._per_input_losses` dictionary is `None`, it's not exactly "per input" anymore, so the next time you'd compute losses on a new tensor, it will add all the regularization multiple times.

**Comment by TimZaman** [comment]
Also confirmed to occur doing:
```
model = keras.models.load_model(model_file)
pred = model(images)
```
The model `model.losses` now contains each regularisation loss twice.
